object-store/src/main/java/org/hypertrace/config/objectstore/ScopeResolutionIdentifiedObjectStoreWithFilter.java

```diff
@@ -0,0 +1,109 @@
+package org.hypertrace.config.objectstore;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.hypertrace.config.service.change.event.api.ConfigChangeEventGenerator;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+/**
+ * ScopeResolutionIdentifiedObjectStoreWithFilter is an abstraction over {@link
+ * IdentifiedObjectStoreWithFilter} to allow scope resolution of fetched configs
```

Comment:
I would have expected this to be an abstraction over DefaultObjectStore - typically if we're resolving scopes, we're not deal with IDs (think something like the booleans associated with detection policies). Or do we want to make that a requirement going forward? Even if we do, we'd still never likely fetch more than one at a time, so I don't think this inheritance relationship would make sense. will read on.

Reply:
but without contexts how are the configs stored? they still have contexts - scopes define the ids in most cases.. i default-store, there is a single config per resource, I don't think that would work multiple scopes right?

Reply:
I think the distinction is that they don't have explicit IDs, but the scope would end up being used as the context. I'm not positive that's needed/wanted as you indicated, so we can skip for now and come back if we find some use case.

Reply:
yeah for now, I didn't see any req as such so skipping..